### PR TITLE
[webcrawler] - fix: improve IP address privacy check logic

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

Fixing the `isPrivateIp` function as it was detecting `10x.` IPs as private while we only want to consider `10.x` as private.

 - Adjusted private IP prefixes to include trailing periods for accurate matching.

### CODE REVIEW

**Review by lasryaric — APPROVED** [review]
LGTM.

**Review by spolu — COMMENTED** [review]
Very high blast radius. But I believe LGTM

Can you get another reviewer on it before merging

**Review by spolu — APPROVED** [review]

**Review by fontanierh — APPROVED** [review]

**Pull request merged**
